src/commonMain/kotlin/fr/acinq/lightning/channel/InteractiveTx.kt

```diff
@@ -829,44 +837,54 @@ data class InteractiveTxSigningSession(
             localCommitmentIndex: Long,
             remoteCommitmentIndex: Long,
             commitTxFeerate: FeeratePerKw,
-            remotePerCommitmentPoint: PublicKey
+            remotePerCommitmentPoint: PublicKey,
+            previousHtlcs: Set<DirectedHtlc>
```

Comment:
Also renamed in [9aa3be5](https://github.com/ACINQ/lightning-kmp/pull/568/commits/9aa3be57e10fadd8fa0e3db1d8e21cdb0255d8e5).

I don't understand what you mean about just being the local ones. It should be everything in the local node spec, right? It's currently initialized with `parentCommitment.localCommit.spec.htlcs`. This includes both incoming and outgoing htlcs and will be identical to the remote node spec but, as you said, reversed.

Reply:
> This includes both incoming and outgoing htlcs and will be identical to the remote node spec but, as you said, reversed.

The htlcs from `parentCommitment.localCommit.spec` and `parentCommitment.remoteCommit.spec` are the same, but reversed. Since we create HTLC transactions based on that, for our _local_ commit, it's important to use the HTLCs from the local spec, otherwise we would generate (invalid) HTLC-success transactions instead of HTLC-timeout transactions, and the opposite, because we would treat incoming HTLCs as outgoing and the opposite.

What I meant is that it's important that we consistently use what's coming from the `localCommit.spec`, not the `remoteCommit.spec`.

Reply:
Ok - gotcha. So that just supports the name change to `localHtlcs`. We can't prevent a caller from accidentally passing the htlcs from the `remoteCommit.spec`, but that would atleast cause the closing tests to fail.